vowpalwabbit/parse_args.cc

```diff
@@ -1290,124 +1290,126 @@ VW::LEARNER::base_learner* setup_base(options_i& options, vw& all)
 
 }
 
-/*
-void register_reduction(vw& all, std::tuple<std::string, VW::LEARNER::base_learner* (*)(VW::config::options_i&, vw&)> tup)
+void register_reductions(vw& all, std::vector<VW::LEARNER::base_learner* (*)(VW::config::options_i&, vw&)>& reductions)
 {
+  std::map<VW::LEARNER::base_learner* (*)(VW::config::options_i&, vw&), std::string> allowlist = {
```

Comment:
some members on this list could be refactored to avoid them being exceptions to the general rule